data_runner/README.md

```diff
@@ -0,0 +1,41 @@
+
+Why wouldn't you do that? I think `system.mem.committed_as` is some pretty big numbers; it's nice to have a game with a score of 1.4×10¹¹.
+
+_Or even make up my own fictional metric and use the [Metrics Submission API](https://docs.datadoghq.com/api/latest/metrics/#submit-metrics) to send in really big numbers?_
+
+See, _now_ you're gettin' it. Go wild.
+
+### How it works, or: the techie bit
+
+This is a [Datadog App](https://github.com/DataDog/apps) that provides a dashboard widget. It uses the standalone version of the [UI Extensions SDK](https://www.npmjs.com/package/@datadog/ui-extensions-sdk) ([ui-extensions-sdk.min.js](https://github.com/stuartlangridge/data-runner/blob/main/ui-extensions-sdk.min.js) in this repo, built from [the upstream github repo](https://github.com/DataDog/ui-extensions-sdk)). For a simpler sort of widget which does less work (and so may show off the required file structures more clearly) it may be useful to examine [dd-app-plain-html-grafana](https://github.com/stuartlangridge/dd-app-plain-html-grafana) or Datadog's own [starter kit](https://github.com/DataDog/starter-kit) for a React app.
+
+On startup, it uses the [widget API client](https://github.com/DataDog/apps/blob/master/docs/en/programming-model.md#api-access) to query the [public Datadog API](https://docs.datadoghq.com/api/) to get a list of your metrics (which it can do because it has the `metrics_read` scope), and then dynamically updates its [widget settings menu](https://github.com/DataDog/apps/blob/master/docs/en/programming-model.md#widget-settings-menu) to list the metrics.
+
+When you choose a metric (either on first adding the widget to your dashboard, or at any time later with the edit button), the Datadog dashboard fires a [`dashboard_custom_widget_options_change`](https://github.com/DataDog/apps/blob/master/docs/en/programming-model.md#widget-settings-menu) custom event which the runner code listens for and records the metric name.
+
+Then periodically the runner polls the [metrics query API](https://docs.datadoghq.com/api/latest/metrics/#query-timeseries-points) for that chosen metric and takes the highest value since the last time it asked, and adds that to the runner's score.
+
+The imagery is all from the fantastic [Kenney](https://kenney.nl/assets/isometric-library-tiles), and the font is from [ooqq](https://github.com/OOQQ/ZX-Spectrum-font).
+
+## Setup
+
+1. In your Datadog account, navigate to Dashboards. Select the dashboard that you would like to add the Game to, or [create a new dashboard](https://docs.datadoghq.com/dashboards/#new-dashboard).
+
+2. In the dashboard, click **+Add Widgets** to the right of the dashboard title. Scroll to the right through the widgets, and then drag and drop the **The data runner** widget into the desired position on your dashboard. Choose a metric you want the Data Runner to find, then click save.
```

Comment:
"Scroll to the right through the widgets, and then..." This looks like instructions for the old widget tray. In the new widget tray you would have to scroll down.